Add base style file for consumption via web components

jshcrowthe opened the pull request:
# PR Summary
@lambertsteven this is the PR we discussed about getting Web Component support in the styleguide.

## Pieces:
- Add base.styl file that is compiled to dist
- Add `$limit-base-styles` flag to remove classes from typography in the base styles
- Add `@font-face` definintions to the fonts section of `_variables.styl`
- Add `fs-styles.html` file to the root of the project and to `bower.json`
- Rev version to `v1.8`

lambertsteven commented on assets/css/elements/_typography.styl:
```diff
@@ -88,63 +88,121 @@ body {
-  text-transform: inherit;
-  margin-top: 0;
-}
+if $limit-base-styles
```
are the classes so much of a bad thing in the base styles that the duplication in code is less problematic?

jshcrowthe replied:
It's just to ensure that if someone *happens* (however unlikely) to use that exact class name in the component to do an override that the base styles.

lambertsteven commented:
I'll pull if you remove the conditional variable (to remove code duplication in the stylus file) and remove the `@font-face` rules. If we provide the font-face rules in the styles, we allow 3rd party users to use our fonts off our CDN, and our fonts are specifically licensed for our use. We'll have to add the font-face rules into theme-engage so only our site uses them.

jshcrowthe commented:
Alright, I will go through and remove the large duplication block and see if I can make that smaller. I do think that having the conditional there is valuable. As to the fonts is that just something that our components are enhanced by being on our site (and therefore having the font available) and other places they just default to the fallback?

lambertsteven commented:
Ya, the components would be enhanced by being on our site.

jshcrowthe commented:
@lambertsteven I have updated the code. I still stand by the conditionals being there but I agree with your point that we shouldn't have to edit code in two places. I have updated the `_typography.styl` to use mixins for each style grouping.
